CSharp/core-DirectLineWebSockets/README.md

```diff
@@ -0,0 +1,136 @@
+
+Each conversation on the Direct Line channel must be explicitly started using the `DirectLineClient.Conversations.StartConversationAsync`.
+Check out the client's [Program.cs](DirectLineClient/Program.cs#L27-L29) class which creates a new `DirectLineClient` and starts a new conversation.
+
+
+````C#
+DirectLineClient client = new DirectLineClient(directLineSecret);
+            
+var conversation = await client.Conversations.StartConversationAsync();
+````
+
+User messages are sent to the Bot using the Direct Line Client `Conversations.PostActivityAsync` method using the `ConversationId` generated in the previous step.
+
+````C#
+while (true)
+{
+    string input = Console.ReadLine().Trim();
+
+    if (input.ToLower() == "exit")
+    {
+        break;
+    }
+    else
+    {
+        if (input.Length > 0)
+        {
+            Activity userMessage = new Activity
+            {
+                From = new ChannelAccount(fromUser),
+                Text = input,
+                Type = ActivityTypes.Message
+            };
+
+            await client.Conversations.PostActivityAsync(conversation.ConversationId, userMessage);
+        }
+    }
+}
+````
+
+Messages from the Bot are being received using WebSocket protocol. For this, after the conversation was created a `streamUrl` is also returned and it will be the target for the WebSocket connection. Check out the client's WebSocket initialization in [Program.cs](DirectLineClient/Program.cs#L31-L34). 
+
+> Note: In this project we use [websocket-sharp](https://github.com/sta/websocket-sharp) to implement the receiver client but as we have the WebSocket URL this functionality can be implemented according to the needs of the solution.
+
+````C#
+using(var webSocketClient = new WebSocket(conversation.StreamUrl))
+{
+    webSocketClient.OnMessage += WebSocketClient_OnMessage;
+    webSocketClient.Connect();
+````
+
+We use the [WebSocketClient_OnMessage](DirectLineClient/Program.cs#L62) method to handle the OnMessage event of the `webSocketClient` witch occurs when the WebSocket receives a message.
+
+````C#
+private static void WebSocketClient_OnMessage(object sender, MessageEventArgs e)
+{
+    var activitySet = JsonConvert.DeserializeObject<ActivitySet>(e.Data);
+    var activities = from x in activitySet.Activities
+        where x.From.Id == botId
+        select x;
+````
+
+DirectLine v3.0 (unlike version 1.1) has support for Attachments (see [Adding Attachments to a Message](https://docs.botframework.com/en-us/core-concepts/attachments) for more information about attachments). Check out the `WebSocketClient_OnMessage` method in [Program.cs](DirectLineClient/Program.cs#L73-L90) to see how the Attachments are retrieved and rendered appropriately based on their type.
```

Comment:
In printed text, Direct Line is normally two words.

It's OK to collapse them in filenames, though. ("DirectLineClient.cs" is OK)